tb_plugin/torch_tb_profiler/plugin.py

```diff
@@ -206,12 +215,32 @@ def kernel_table_route(self, request):
 
     @wrappers.Request.application
     def trace_route(self, request):
+        def build_trace_counter(gpu_id, start_time, counter_value):
+            util_json = ", {{\"ph\":\"C\", \"name\":\"GPU {} Est. SM Efficiency\", " \
+                        "\"pid\":{}, \"ts\":{}, " \
+                        "\"args\":{{\"Est. SM Efficiency\":{}}}}}".format(
+                gpu_id, gpu_id, start_time, counter_value
+            )
+            return util_json
         name = request.args.get("run")
         worker = request.args.get("worker")
 
         run = self._get_run(name)
         profile = run.get_profile(worker)
         raw_data = self._cache.read(profile.trace_file_path)
+        util_bytes = profile.gpu_util_json
+
+        counter_json_str = ""
+        for gpu_id in range(len(profile.approximated_sm_efficency_ranges)):
+            ranges = profile.approximated_sm_efficency_ranges[gpu_id]
+            for r in ranges:
+                efficiency_json_start = build_trace_counter(gpu_id, r[0][0], r[1])
+                efficiency_json_finish = build_trace_counter(gpu_id, r[0][1], 0)
+                counter_json_str += (efficiency_json_start + efficiency_json_finish)
+        counter_json_bytes = bytes(counter_json_str, 'utf-8')
```

Comment:
in case of the raw data is gz file, I think the GPU SM efficiency and utilization will not be enabled?

Reply:
Cool! It's a big bug. Thanks a lot!
If it is gz file, I have to decompress it firstly, then append these metrics to the end, and compress it, and add an "update" member function in io.Cache to save the compressed result into local file?
BTW, if the original data file is not "gz", maybe we can save the compressed data into local file and update io.Cache for following calls.